bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -358,7 +358,7 @@ public synchronized void writeComplete(int rc, long ledgerId, long entryId, Book
             } else {
                 LOG.warn("Failed to write entry ({}, {}) to bookie ({}, {}): {}",
                         ledgerId, entryId, bookieIndex, addr, BKException.getMessage(rc));
-                lh.handleBookieFailure(ImmutableMap.of(bookieIndex, addr));
+                lh.executeOrdered(() -> lh.handleBookieFailure(ImmutableMap.of(bookieIndex, addr)));
```

Comment:
We shouldn't change the thread here, it will delay the handleBookieFailure process. Please check the failed test `LedgerClose2Test#testMetadataChangedDuringClose`

Reply:
Gotcha!